docs/workflow.md

```diff
@@ -21,6 +21,21 @@ This repo has bots that manage all stale PRs. Stale PRs will be autoclosed.
 - *Do* use GitHub's tooling. Re-request review after all feedback has been addressed.
 - *Do* pay special attention to the commit message. Ensure the merge message is appropriate and helpful to the future reader. See [merge commit considerations](#merge-commit-considerations).
 
+#### Pull Request Etiquette and Best Practices
+
+- If you disagree with the overall approach of the PR, comment on the general PR rather than individual lines of code.
+- Leaving [suggested changes](https://docs.github.com/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request) is welcomed, but please never commit them for a PR you did not create.
+- When you are seeking to understand something rather than request corrections, it is suggested to use language such as "I'm curious ..." as a prefix to comments.
+- For comments that are just optional suggestions or are explicitly non-blocking, prefix them with "nit: " or "non-blocking: ".
+- When to mark a PR as "Approved"
+  - You feel confident that the code meets a high quality bar, has adequate test coverage, is ready to merge.
+  - You have left comments that are uncontroversial and there is a shared understanding with the author that the comments can be addressed or resolved prior to being merged without significant discussion or significant change to the design or approach.
```

Comment:
You need to approve the latest version, so this probably could use some workshopping to make the current process.

Reply:
I think their meaning was, if it's trivial, you can approve and mention a nit or small request.
Is the suggestion to state the flip side of that and state we must ultimately approve without **any** follow-up actions? Do we not allow trivial/nitpicks in an approval?

Reply:
I'm referring to `shared understanding with the author that the comments can be addressed or resolved prior to being merged ` 

I don't think that's referring to nits only but rather to changes that are not controversial but need addressed. 

For example, using NoAllocatorNumerator on an IList in a hot path in restore. That's something that needs addressed, but I trust that a person would do it without needing re-review. The EO meaning that latest iteration needs to be approved means that if a person addresses your feedback, they might *need* your approval again. 

It may be  as simple as we need to mention that the latest iteration of the PR needs approved.

Reply:
> It may be as simple as we need to mention that the latest iteration of the PR needs approved.
It sounds like we agree. This is basically paraphrasing of my:

>  ..and state we must ultimately approve without any follow-up actions? 

It would make it less ambiguous since today we aren't very consistent with approvals. I'm personally OK with using judgement and trust, as you mention, but also sometimes it's better to just have a simple, yet hard guideline.

Reply:
I'd prefer something that's not strict, just make it match what we have to do since that doc was obviously written prior the EO regarding approval of latest PR iteration.

Reply:
Mind proposing a suggestion that I can commit?

Reply:
```suggestion
  - You have left comments that are uncontroversial and there is a shared understanding with the author that the comments can be addressed or resolved either within this PR or a follow up PR. without a significant change to the design or approach. If the author addresses within the same, you may need to re-approve the PR as the latest iteration of a PR needs is enforced.
 ```
 
 How's this?